data_diff/tracking.py

```diff
@@ -0,0 +1,114 @@
+#
+# This module contains all the functionality related to the anonymous tracking of data-diff use.
+#
+
+import toml
+import logging
+import os
+import json
+import platform
+from time import time
+from typing import Any, Dict, Optional
+import urllib.request
+from uuid import uuid4
+
+TRACK_URL = "https://api.perfalytics.com/track"
+START_EVENT = "os_diff_run_start"
+END_EVENT = "os_diff_run_end"
+TOKEN = "ccb8c3a6-3b6f-445c-ad67-994efa7bd020"
+
+DEFAULT_PROFILE = os.path.expanduser("~/.datadiff.toml")
+
+
+def _load_profile():
+    try:
+        with open(DEFAULT_PROFILE) as f:
+            conf = toml.load(f)
+    except FileNotFoundError as e:
+        conf = {}
+
+    if "anonymous_id" not in conf:
+        conf["anonymous_id"] = str(uuid4())
+        with open(DEFAULT_PROFILE, "w") as f:
+            toml.dump(conf, f)
```

Comment:
Just to clarify: this file is not used for anything but for datadiff's internal state, correct?

(Why? => This operation will reformat the config and will remove all comments from it. If the config is crafted by humans, they will be unhappy. If it is machine-generated, there is no problem.)

Reply:
Currently it is only used by the machine. I was thinking of making it a "global" config, in a future PR.

If you have an idea on how to preserve human-happiness artifacts, I'll be happy to include it.

Reply:
There is no way to preserve that. I would suggest using another file with the anonymous id in that case (e.g. `~/.datadiff-id.toml`).